delivery_carrier_preference_release_operations/models/stock_move.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import models
+
+
+class StockMove(models.Model):
+
+    _inherit = "stock.move"
+
+    def release_available_to_promise(self):
+        existing_backorders = {
+            pick.id: pick.mapped("backorder_ids").ids
+            for pick in self.mapped("picking_id")
+        }
+        res = super().release_available_to_promise()
```

Comment:
This step will apply the pull rules and the pull rules depend on the carrier.
The carrier must be computed based on what is available to promise and updated before the call to super